### ISSUE TITLE
Filed to generate class based on enum

### ISSUE THREAD

**Issue by andreikrutsko** [graphql-java-generator/graphql-maven-plugin-project#202]
Failed to generate 

schema:

```
type UserUpdateNotification {
  title: String
  body: String!
  class: UserUpdateNotificationClass!
  reference_nr: String
}

enum UserUpdateNotificationClass {
  success
  error
  info
}
```

output:
`
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:3.11.0:compile (default-compile) on project ***: Compilation failure
[ERROR] /***/client/UserUpdateNotification.java:[99,81] getClass() in com.***.client.UserUpdateNotification cannot override getClass() in java.lang.Object
[ERROR]   overridden method is final
`

**Comment by etienne-sf** [comment]
Arg, yes. 

As the schema has a ´class´ attribute, the code generates a ´getClass()´ method. 

I'll correct that, by protecting standard Object's ´_getXxxx()´ methods, by replacing them by ´getXxxx()´. 

In the meanwhile, you can (if possible) rename the ´class´ attribute by ´_class´ or ´class2´. 

Étienne

**Comment by andreikrutsko** [comment]
thats part of a predefined schema provided by a resource supplier 
we are not allowed to modify, just have to get "as is" and build a backend client
in the moment GraphQL server feeds data to numerous GUI frontends and any changes forbidden

is there any way to avoid such exceptional case in generator codebase?

**Comment by etienne-sf** [comment]
Hello,

There is only one getter that could generate this error, for the `class` attribute.

The generated methods `getClass` and `setClass` are now generated as `get_Class` and `set_Class` .

This correction is available in the new release: 2.3.2

Etienne

**Comment by andreikrutsko** [comment]
works!
many thanks!
